aten/src/ATen/core/jit_type.h

```diff
@@ -1558,6 +1572,9 @@ struct CAFFE2_API ClassType : public NamedType {
   // available from c10
   std::vector<std::string> attributeNames_;
   std::vector<TypePtr> attributeTypes_;
+  // Mapping of constant names -> their value.
+  std::vector<std::string> constantNames_;
```

Comment:
This creates two namespaces for constants and slots, which allows there to be both a slot `foo` and a constant `foo`. We do not want that, so we need to add checks that prevent these namespaces from overlapping.